ffmpeg/lpms_ffmpeg.c

```diff
@@ -726,20 +945,40 @@ int encode(AVCodecContext* encoder, AVFrame *frame, struct output_ctx* octx, AVS
   char errstr[AV_ERROR_MAX_STRING_SIZE] = {0}; \
   if (!ret) { fprintf(stderr, "should not happen\n"); ret = AVERROR(ENOMEM); } \
   if (ret < -1) av_strerror(ret, errstr, sizeof errstr); \
-  fprintf(stderr, "%s: %s", msg, errstr); \
+  fprintf(stderr, "%s: %s\n", msg, errstr); \
   goto encode_cleanup; \
 }
 
+  int ret = 0;
   AVPacket pkt = {0};
 
   if (AVMEDIA_TYPE_VIDEO == ost->codecpar->codec_type && frame) {
     octx->res->frames++;
     octx->res->pixels += encoder->width * encoder->height;
+    if (!octx->nb_frame) {
+      //fprintf(stderr, "Forcing IFrame\n");
+      frame->pict_type = AV_PICTURE_TYPE_I;
```

Comment:
IIUC this forces the first frame of each segment encoded to be an I-frame? Was this not the case before?

Reply:
If the encoder is persistent across sessions, we needed some way to tell it to insert an I-frame for the beginning of a segment. But with the [flush patch](https://github.com/livepeer/lpms/pull/141/files#diff-5e694afe073853397a89ba4797aad0d6R974-R977) [1] that was just recently re-introduced, this might not be needed anymore. Need to double check that.

[1] Doh, just noticed the almost-identical code block below. Fixed locally.

Reply:
This turns out to have been necessary even with the flush patch, so we keep it in, but in slightly modified form using `octx->res->frames` which already tracks the output frame count for the segment.